From agust@denx.de  Wed May  5 11:13:44 2010
From: Anatolij Gustschin <agust@denx.de>
Date: Wed,  5 May 2010 00:18:59 +0200
Subject: serial: mpc52xx_uart: fix null pointer dereference
To: linux-kernel@vger.kernel.org
Cc: linuxppc-dev@ozlabs.org, grant.likely@secretlab.ca, leonardo.lists@gmail.com, gregkh@suse.de, Anatolij Gustschin <agust@denx.de>
Message-ID: <1273011539-13994-1-git-send-email-agust@denx.de>


Commit 6acc6833510db8f72b5ef343296d97480555fda9
introduced NULL pointer dereference and kernel crash
on ppc32 machines while booting. Fix this bug now.

Reported-by: Leonardo Chiquitto <leonardo.lists@gmail.com>
Tested-by: Leonardo Chiquitto <leonardo.lists@gmail.com>
Signed-off-by: Anatolij Gustschin <agust@denx.de>
Cc: stable <stable@kernel.org>
Acked-by: Grant Likely <grant.likely@secretlab.ca>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/serial/mpc52xx_uart.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/serial/mpc52xx_uart.c
+++ b/drivers/serial/mpc52xx_uart.c
@@ -1467,7 +1467,7 @@ mpc52xx_uart_init(void)
 	/*
 	 * Map the PSC FIFO Controller and init if on MPC512x.
 	 */
-	if (psc_ops->fifoc_init) {
+	if (psc_ops && psc_ops->fifoc_init) {
 		ret = psc_ops->fifoc_init();
 		if (ret)
 			return ret;
